src/config.ts

```diff
@@ -166,6 +178,7 @@ export interface Config {
  * user-provided value will be used to extend the default value.
  */
 export const defaultConfig = {
+  clsMechanism: 'auto' as ContextPropagationMechanism,
```

Comment:
make `ContextPropagationMechanism` an enum and avoid the cast.

Reply:
I don't think user-facing values can be enums. It would explicitly prohibit them from specifying string parameters unless they casted it.